optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -220,8 +220,8 @@ public void solvingStarted(DefaultSolverScope<Solution_> solverScope) {
 
     @Override
     public void solvingEnded(DefaultSolverScope<Solution_> solverScope) {
-        super.solvingEnded(solverScope);
         solverScope.endingNow();
+        super.solvingEnded(solverScope);
```

Comment:
This makes me nervous. The lifecycle event calls have a very predictable order: first superclasses, then subclasses.

The good news is we can have our cake and eat it too:
Both PartionedSolver and AstractSolver call solverScope.edningNow() immediatly after calling the super method, so it can be moved into the super method. I must have looked over that when splitting up the 2 solver implementations.
Once it's there, we can discuss moving it. 

With brings open the side-effect that if we move it up, any actions after the calling endingNow no longer affect the endingSystemTimeMillis... but the contract with the user is that they should.

Reply:
Let's discuss real-time next week or after VRP meeting.

Reply:
Either way, the decision is made please lmk. If we can have this PR in, then I can update https://github.com/kiegroup/droolsjbpm-integration/pull/1776 to include score calculation speed metric.

Reply:
I agree with @ge0ffrey that the change might impact the "contract" of the lifecycle methods. On the other hand it doesn't make sense to fire events that are not in a consistent state. I hope we'll be able to find a solution this Friday.